packages/elements/src/components/ino-markdown-editor/markdown-serializer.ts

```diff
@@ -0,0 +1,132 @@
+import Paragraph from '@tiptap/extension-paragraph';
+import OrderedList from '@tiptap/extension-ordered-list';
+import BulletList from '@tiptap/extension-bullet-list';
+import ListItem from '@tiptap/extension-list-item';
+import CodeBlock from '@tiptap/extension-code-block';
+import HardBreak from '@tiptap/extension-hard-break';
+import Link from '@tiptap/extension-link';
+
+// Source taken from
+// https://gitlab.com/gitlab-org/gitlab/-/blob/master/app/assets/javascripts/content_editor/services/markdown_serializer.js
+const defaultSerializerConfig = {
+  marks: {
+    em: defaultMarkdownSerializer.marks.em,
+    [Bold.name]: defaultMarkdownSerializer.marks.strong,
+    [Code.name]: defaultMarkdownSerializer.marks.code,
+    [Italic.name]: {
+      open: '_',
+      close: '_',
+      mixable: true,
+      expelEnclosingWhitespace: true,
+    },
+    [Link.name]: defaultMarkdownSerializer.marks.link,
+    [Strike.name]: {
+      open: '~~',
+      close: '~~',
+      mixable: true,
+      expelEnclosingWhitespace: true,
+    },
+  },
+  nodes: {
+    ...defaultMarkdownSerializer.nodes,
+    [Blockquote.name]: (state, node) => {
+      if (node.attrs.multiline) {
+        state.write('>>>');
+        state.ensureNewLine();
+        state.renderContent(node);
+        state.ensureNewLine();
+        state.write('>>>');
+        state.closeBlock(node);
+      } else {
+        state.wrapBlock('> ', null, node, () => state.renderContent(node));
+      }
+    },
+    [CodeBlock.name]: defaultMarkdownSerializer.nodes.code_block,
+    [Heading.name]: defaultMarkdownSerializer.nodes.heading,
+    [HorizontalRule.name]: defaultMarkdownSerializer.nodes.horizontal_rule,
+    [BulletList.name]: defaultMarkdownSerializer.nodes.bullet_list,
+    [OrderedList.name]: defaultMarkdownSerializer.nodes.ordered_list,
+    [ListItem.name]: defaultMarkdownSerializer.nodes.list_item,
+    [Paragraph.name]: defaultMarkdownSerializer.nodes.paragraph,
+    [Text.name]: defaultMarkdownSerializer.nodes.text,
+  },
+};
+
+// Source taken from
+// prosemirror-markdown/src/from_markdown.js
+const {
+  bullet_list,
+  ordered_list,
+  link,
+  heading,
+  fence,
+} = (defaultMarkdownParser.tokens as unknown) as { [key: string]: TokenConfig };
+
+const defaultParserTokens: { [key: string]: TokenConfig } = {
+  blockquote: { block: Blockquote.name },
+  paragraph: { block: Paragraph.name },
+  list_item: { block: ListItem.name },
+  bullet_list: {
+    ...bullet_list,
+    block: BulletList.name,
+  },
+  ordered_list: {
+    ...ordered_list,
+    block: OrderedList.name,
+  },
+  heading,
+  code_block: { block: CodeBlock.name, noCloseToken: true },
+  fence: {
+    ...fence,
+    block: CodeBlock.name,
+  },
+  hr: { node: HorizontalRule.name },
+  hardbreak: { node: HardBreak.name },
+
+  [Italic.name]: { mark: Italic.name },
+  em: { mark: Italic.name },
+  [Bold.name]: { mark: Bold.name },
+  strong: { mark: Bold.name },
+  link,
```

Comment:
Clicking the link action in preview mode seems to have no effect.
Also, writing something like `[Test](www.google.de)` will not be parsed in preview mode.
Works when writing in markdown and switching to preview, though.